src/taipy/core/pipeline/pipeline.py

```diff
@@ -260,3 +263,45 @@ def submit(
         from ._pipeline_manager_factory import _PipelineManagerFactory
 
         return _PipelineManagerFactory._build_manager()._submit(self, callbacks, force, wait, timeout)
+
+    @classmethod
+    def _to_model(cls, pipeline):
+        datanode_task_edges = defaultdict(list)
+        task_datanode_edges = defaultdict(list)
+
+        for task in pipeline._get_tasks().values():
+            task_id = str(task.id)
+            for predecessor in task.input.values():
+                datanode_task_edges[str(predecessor.id)].append(task_id)
+            for successor in task.output.values():
+                task_datanode_edges[task_id].append(str(successor.id))
+        return _PipelineModel(
+            pipeline.id,
+            pipeline.owner_id,
+            list(pipeline._parent_ids),
+            pipeline.config_id,
+            pipeline._properties.data,
+            cls.__to_task_ids(pipeline._tasks),
+            _utils._fcts_to_dict(pipeline._subscribers),
+            pipeline.version,
+        )
+
+    @classmethod
+    def _from_model(cls, model):
+        return Pipeline(
+            model.config_id,
+            model.properties,
+            model.tasks,
+            model.id,
+            model.owner_id,
+            set(model.parent_ids),
+            [
+                _Subscriber(_utils._load_fct(it["fct_module"], it["fct_name"]), it["fct_params"])
+                for it in model.subscribers
+            ],  # type: ignore
+            model.version,
+        )
```

Comment:
Why do we remove the `except NonExistingTask:` and `except KeyError:` compare to the logic in _pipeline_repository.py?

Reply:
because the model is storing a list of TaskId, and we're creating Pipeline using TaskId from the model without getting any actual Task, so the exception is not necessary 😃